pyuvdata/uvdata/corr_fits_src/corr_fits.pyx

```diff
@@ -121,3 +124,124 @@ cpdef numpy.ndarray[ndim=1, dtype=numpy.float64_t] get_cable_len_diffs(
+  cdef numpy.float64_t[:, ::1]  j_ind = ind1 / sig1
+  cdef numpy.float64_t[:, ::1]  k_ind = ind1 / sig2
+
+  cdef Py_ssize_t i, j, k, l
+
+  if x.size > 800:  # pragma: nocover
+    with nogil, parallel():
+      for j in prange(x.shape[1]):
+        for i in range(x.shape[0]):
+          for l in range(k_ind.shape[0]):
+            for k in range(j_ind.shape[0]):
+              _khat[i, j] += (
+                1./ (pi * sqrt( 1 - x[i, j] ** 2)) * (
+                  exp(-1. / (2 * (1 - x[i,j] ** 2)) * (j_ind[k, j] ** 2 + k_ind[l, j] ** 2 - 2 * x[i, j] * j_ind[k, j] * k_ind[l,j]))
+                  + exp(-1. / (2 * (1 - x[i,j] ** 2)) * (j_ind[k, j] ** 2 + k_ind[l, j] ** 2 + 2 * x[i, j] * j_ind[k, j] * k_ind[l,j]))
+                )
+              )
+  else:
+    for i in range(x.shape[0]):
+      for j in range(x.shape[1]):
+        for l in range(k_ind.shape[0]):
+          for k in range(j_ind.shape[0]):
+            _khat[i, j] += (
+              1./ (pi * sqrt( 1 - x[i, j] ** 2)) * (
+                exp(-1. / (2 * (1 - x[i,j] ** 2)) * (j_ind[k, j] ** 2 + k_ind[l, j] ** 2 - 2 * x[i, j] * j_ind[k, j] * k_ind[l,j]))
+                + exp(-1. / (2 * (1 - x[i,j] ** 2)) * (j_ind[k, j] ** 2 + k_ind[l, j] ** 2 + 2 * x[i, j] * j_ind[k, j] * k_ind[l,j]))
+              )
+            )
+
+  return khat
+
+
+cpdef get_khat(rho, sig1, sig2):
+    rho = np.ascontiguousarray(rho, dtype=np.float64)
+    squeeze = False
+    if rho.ndim == 1:
+      squeeze = True
+      rho = rho.reshape(1, -1)
+    sig1 =  np.ascontiguousarray(sig1, dtype=np.float64)
+    sig2 =  np.ascontiguousarray(sig2, dtype=np.float64)
+
+    khat = np.zeros((rho.shape[0], rho.shape[1]), dtype=np.float64)
+    khat = _compute_khat(
+        rho, sig1, sig2
+    )
+
+    if squeeze:
+      khat = khat.squeeze()
+    return khat
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef numpy.ndarray[ndim=2, dtype=numpy.float64_t[:, ::1]] _get_cheby_coeff(
+    numpy.ndarray[ndim=3, dtype=numpy.float64_t] rho_coeff,
+    numpy.ndarray[ndim=1, dtype=numpy.int64_t] sv_inds_right1,
+    numpy.ndarray[ndim=1, dtype=numpy.int64_t] sv_inds_right2,
+    numpy.ndarray[ndim=1, dtype=numpy.float64_t] ds1,
+    numpy.ndarray[ndim=1, dtype=numpy.float64_t] ds2
+):
+  cdef int i
+  cdef int j
+  cdef int n = len(ds1)
+  cdef numpy.ndarray[ndim=2, dtype=numpy.float64_t] t = np.zeros((n, 3), dtype=np.float64)
+
+  for i in cython.parallel.prange(n, nogil=True):
+      for j in range(3):
+          t[i, j] = 1e4 * (
+              rho_coeff[(sv_inds_right1[i] - 1), (sv_inds_right2[i] - 1), j] * ds1[i] * ds2[i]
+              + rho_coeff[(sv_inds_right1[i] - 1), sv_inds_right2[i], j] * ds1[i] * (0.01 - ds2[i])
+              + rho_coeff[sv_inds_right1[i], (sv_inds_right2[i] - 1), j] * (0.01 - ds1[i]) * ds2[i]
+              + rho_coeff[sv_inds_right1[i], sv_inds_right2[i], j] * (0.01 - ds1[i]) * (0.01 - ds2[i])
+          )
+
+  return t
```

Comment:
same comment here about the compilation. Good to verify we get the best we can out of cython.